### ISSUE TITLE
Gargoyle builds on recent macOS do not display text

### ISSUE THREAD

**Issue by townba** [garglk/garglk#903]
OS: macOS 15.0.1 (24A348)

For the past several months, Gargoyle builds do not display text, but the released app bundle works. The games start and run — I can even get the violin to play in Infocom's "Sherlock" — but nothing is displayed.

Things I've tried that haven't worked:
* Trying it on two different MacBooks Pro (both M2)
* Using both MacPorts and Homebrew libraries
* Removing my `.bashrc` and `.garglkrc`
* Switching to older Gargoyle source that I know worked for me in the past

One thing I haven't tried is using old versions of the libraries provided by MacPorts and Homebrew.

Is it just me? If so, any ideas on what I'm doing wrong?

**Comment by cspiegel** [comment]
The only Mac I've currently got access to is an Intel Monterey (and it's in a VM); and I'm as "brew updated" as possible, with some failures there as Homebrew no longer supports it. That being said, Gargoyle looks fine on there for me.

I'll try to get my hands on a modern Mac and see what's going on, but I've always just kind of coaxed that Mac code along as I've made changes to Gargoyle: I really have no experience in Mac development so it may take some effort to figure things out.

That said.. I've always had a long-term plan to ditch the Mac-specific UI code for Qt. You can already build with a Qt interface on Mac, but it doesn't behave exactly like a Mac program, and I haven't yet been able to get it bundled up into a DMG.  I'll open a ticket for that, at least, so it doesn't get totally lost. It may well wind up being easier to improve Qt on Mac than to fix whatever issue you're seeing...

Although for testing purposes, you ought to be able to build and run the Qt version. I just tested it in my VM and it still builds and runs. So if you're looking to test Gargoyle stuff that's not specific to the Mac interface, at least you'd be able to do that.

**Comment by cspiegel** [comment]
Added #904 to keep the Mac Qt port on the radar.

**Comment by townba** [comment]
Yep, the Qt version builds and runs just fine. (I mean, I tried "Ballyhoo" and "Bureaucracy" for no more than 30 seconds, but they worked.) Thanks for the advice! If I get some time, I'll see if I can modify `gargoyle_osx.sh` to work with the Qt build.

**Comment by cspiegel** [comment]
Update here: I tried on a new M3 MacBook Pro with MacOS 15.1, and the Cocoa build works for me.

This is just from running `gargoyle_osx.sh` and running the resulting bundle.  The graphics are fine... so I guess that's a data point, but unfortunately not particularly useful for you.

**Comment by angstsmurf** [comment]
This might not be helpful, but it is another data point: `gargoyle_osx.sh` seems to stall indefinitely after printing `[100%] Built target frankendrift` on my M1 MacBook Air, also running macOS Sequoia  15.1.

Perhaps there is some dependency I haven't installed properly.

**Comment by cspiegel** [comment]
> EDIT: Seems to have fixed itself somehow. It built just fine the second time I tried. Sorry for the noise!

Not noise at all! This is actually something I periodically run into on Linux (and only Linux), but only in my "main" user account.

I have a build environment set up that runs as a different user and automates thousands of builds of various packages, including Gargoyle, and I've _never_ seen it there, so I always just kind of assumed I had some bizarre thing going on. It seems to only affect largely parallel builds: I normally run either with `-j24` or just `-j` by itself.  And rebuilding "solves" the problem: it makes it past the last step without issue.

Anyway, I've never seen it on Mac, nor do I _recall_ seeing it while doing MinGW builds on Linux. And... I don't think I've ever seen one of the GitHub runners hit it, either.

So forever I just assumed it was some problem locally, but since you've seen it, I'm guessing there's _some_ weird race condition or similar, though I don't have the first idea how to debug it.

It's really good knowing you're hitting it too, meaning there may actually be some way of fixing it.

**Comment by ReddestDream** [comment]
I can confirm I can reproduce @townba's issue. I can't see any text if I try to build with the latest gargoyle_osx.sh. I'm on an M1 MBA 16 GB RAM macOS 15.3.1. Thanks!

**Comment by alecazam** [comment]
Brew only builds libs and exes as x64 or arm64 - never universal apps.  And they're prepping to delete macOS x64.  If you're on arm64, you need to make sure that you have deleted the old x64 cellar.   And then brew only keeps like 2 versions of everything built.   So if you need to reference older versions, that's just not possible.
